cmd/acra-server/common/listener.go

```diff
@@ -443,16 +488,36 @@ func (server *SServer) StartCommandsFromFileDescriptor(fd uintptr) {
 	if err != nil {
 		logger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCantOpenFileByDescriptor).
 			Errorln("System error: can't start listen for file descriptor")
-		server.errorSignalChannel <- syscall.SIGTERM
 		return
 	}
 	listenerWithFileDescriptor, ok := listenerFile.(network.ListenerWithFileDescriptor)
 	if !ok {
 		logger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorFileDescriptionIsNotValid).
 			Errorf("System error: file descriptor %d is not a valid socket", fd)
+		server.errorSignalChannel <- syscall.SIGTERM
 		return
 	}
```

Comment:
Hm... Why signal TERM on line 497 and 484, but not for `net.FileListener` failure?

Reply:
Yeah. While refactoring, I had no clear understanding why do we need this, but later I came up that we have to gracefully shutdown our service in case of errors during listener creation, since we invoke `Start` (and its variations) in separate goroutine in AcraServer (in contrast to AcraTranslator where we don't and consequently are able to just return). Thank you for attentiveness